### DISCUSSION TITLE
Consider auto-generated changelog

### DISCUSSION THREAD

**Discussion by brettcannon** [brettcannon/python-launcher#157] [Ideas]
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

Not sure if it's worth it to drop scriv, but it may make PRs easier from external contributors as there is less of a chance of someone forgetting a news file.

**Comment by brettcannon** [comment]
https://docs.github.com/en/rest/reference/repos#generate-release-notes-content-for-a-release is the REST API for getting the generated release notes. Looks like it would be easiest to get the release tagged, then generate the release notes, then update the release with those notes.

**Comment by brettcannon** [comment]
Will also have to decide whether to keep `CHANGELOG.md` around or continue to maintain it. Would need to use https://docs.github.com/en/rest/reference/repos#get-the-latest-release to get the last release to base the notes off of.

**Comment by brettcannon** [comment]
https://github.com/cli/cli/issues/4466 is tracking adding this to `gh`.

**Reply by brettcannon** [reply]
This has been fixed and will be available in a future release of `gh`!

https://github.com/cli/cli/pull/4467

**Reply by brettcannon** [reply]
Released in [`gh` 2.4.0](https://github.com/cli/cli/releases/tag/v2.4.0).

**Comment by FollowTheProcess** [comment]
I've used https://github.com/marketplace/actions/release-drafter for this in some of my projects, it takes a small amount of configuring so you can get the format/labels how you like it etc. but once it's set up it truly is a "fire and forget" type thing, just reliably produces nice release notes with every tag!

My only potential gripe is that I end up manually copying and pasting the notes from the "Draft" to the actual release text, I'm sure it's possible to automate this with some API calls but I haven't got around to it yet!

I've not tried the GitHub built in versions yet so I can't offer any kind of comparison

**Comment by brettcannon** [comment]
Tracking in https://github.com/brettcannon/python-launcher/issues/179 .
